packages/venia-concept/src/components/Checkout/paymentsForm.js

```diff
@@ -1,90 +1,230 @@
-                            Payment Method
-                        </Label>
-                        <Select
-                            field="paymentMethod"
-                            initialValue={initialValue}
-                            items={selectablePaymentMethods}
+                    <div className={classes.braintree}>
+                        <BraintreeDropin
+                            isRequestingPaymentNonce={
+                                this.state.isRequestingPaymentNonce
+                            }
+                            onError={this.cancelPaymentNonceRequest}
+                            onSuccess={this.setPaymentNonce}
                         />
                     </div>
+                    <div className={classes.address_check}>
+                        <Checkbox
+                            field="addresses_same"
+                            label="Billing address same as shipping address"
+                        />
+                    </div>
+                    {!formState.values.addresses_same &&
+                        this.billingAddressFields}
                 </div>
                 <div className={classes.footer}>
-                    <Button priority="high" type="submit" disabled={submitting}>
-                        Save
+                    <Button className={classes.button} onClick={this.cancel}>
+                        Cancel
+                    </Button>
+                    <Button
+                        className={classes.button}
+                        priority="high"
+                        type="submit"
+                        disabled={submitting}
+                    >
+                        Use Card
                     </Button>
-                    <Button onClick={this.cancel}>Cancel</Button>
                 </div>
-            </Form>
+            </Fragment>
         );
-    }
+    };
+
+    setFormApi = formApi => {
+        this.formApi = formApi;
+    };
 
+    /*
+     *  Event Handlers.
+     */
     cancel = () => {
         this.props.cancel();
     };
 
-    submit = ({ paymentMethod }) => {
-        const selectedPaymentMethod = this.props.availablePaymentMethods.find(
-            ({ code }) => code === paymentMethod
-        );
+    submit = () => {
+        this.setState({ isRequestingPaymentNonce: true });
+    };
 
-        if (!selectedPaymentMethod) {
-            console.warn(
-                `Could not find the selected payment method ${selectedPaymentMethod} in the list of available payment methods.`
-            );
-            this.cancel();
-            return;
+    setPaymentNonce = value => {
+        this.setState({
+            isRequestingPaymentNonce: false
+        });
+
+        // Build up the billing address payload.
+        const formValue = this.formApi.getValue;
+        let billingAddress = {
+            sameAsShippingAddress: formValue('addresses_same') || false
```

Comment:
I'm guessing this `sameAsShippingAddress` code was intended to prevent you from having to make an async/store request for the shipping address within the component, but I feel like if you managed to pull that to this level it would be a little cleaner. In fact I think it would remove the need for much of the code around `sameAsShippingAddress` checks.

```js
if (formValue('addresses_same') {
  billingAddress = await actions.retrieveShippingAddress()
}
```

Of course this may be asking for something that isn't recommended to do in redux, I'm really not sure.